packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -122,21 +128,31 @@ export async function runBuildAndSubmitAsync(
 
   const customBuildConfigMetadataByPlatform: { [p in AppPlatform]?: CustomBuildConfigMetadata } =
     {};
-  for (const buildProfile of buildProfiles) {
-    validateBuildProfileVersionSettings(buildProfile, easJsonCliConfig);
-    const maybeMetadata = await validateCustomBuildConfigAsync(projectDir, buildProfile.profile);
-    if (maybeMetadata) {
-      customBuildConfigMetadataByPlatform[toAppPlatform(buildProfile.platform)] = maybeMetadata;
-    }
-  }
-
   const startedBuilds: {
     build: BuildWithSubmissionsFragment | BuildFragment;
     buildProfile: ProfileData<'build'>;
   }[] = [];
   const buildCtxByPlatform: { [p in AppPlatform]?: BuildContext<Platform> } = {};
 
   for (const buildProfile of buildProfiles) {
+    const { exp, projectId } = await getDynamicPrivateProjectConfigAsync({
+      env: buildProfile.profile.env,
+    });
+
+    validateBuildProfileVersionSettings(buildProfile, easJsonCliConfig);
+    const maybeMetadata = await validateCustomBuildConfigAsync(projectDir, buildProfile.profile);
+    if (maybeMetadata) {
+      customBuildConfigMetadataByPlatform[toAppPlatform(buildProfile.platform)] = maybeMetadata;
+    }
+    if (buildProfile.profile.channel) {
```

Comment:
why not just move it next to validateBuildProfileConfigMatchesProjectConfigAsync, with this change you are not validating that earlier from the user perspective, just from the code perspective.